src/core/drive/renderer.ts

```diff
@@ -10,10 +10,15 @@ export abstract class Renderer {
   abstract delegate: RenderDelegate
   abstract newBody: HTMLBodyElement
 
-  renderView(callback: RenderCallback) {
-    this.delegate.viewWillRender(this.newBody)
-    callback()
-    this.delegate.viewRendered(this.newBody)
+  renderView(callback: RenderCallback):Promise<void> {
+    return new Promise((resolve:() => void) => {
```

Comment:
@domchristie I believe this should be `return new Promise<void>((resolve) => {`, which would resolve the error you mentioned.

Reply:
Thanks! 🙏